hawkular-bus-common/src/main/java/org/hawkular/bus/common/SimpleMessageProcessor.java

```diff
@@ -67,16 +65,16 @@ public MessageId send(BasicMessage basicMessage, Map<String, String> headers) th
         return super.sendAndListen(producerCtx, basicMessage, responseListener, headers);
     }
 
-    public <R extends BasicMessage> ListenableFuture<BasicMessageWithExtraData<R>> sendRPC(BasicMessage basicMessage,
-            Class<R> expectedResponseMessageClass) throws JMSException {
-        return super.sendRPC(producerCtx, basicMessage, expectedResponseMessageClass);
-    }
-
-    public <R extends BasicMessage> ListenableFuture<BasicMessageWithExtraData<R>> sendRPC(BasicMessage basicMessage,
-            Class<R> expectedResponseMessageClass, Map<String, String> headers)
-            throws JMSException {
-        return super.sendRPC(producerCtx, basicMessage, expectedResponseMessageClass, headers);
-    }
+//    public <R extends BasicMessage> ListenableFuture<BasicMessageWithExtraData<R>> sendRPC(BasicMessage basicMessage,
```

Comment:
The motivation for using ListenableFuture was to support async messaging. I removed it for two reasons. It was not used anywhere. And in a separate PR/ticket I am going to propose a different API for async that utilizes RxJava.